### PR TITLE
Adds Itinerary to UI

### PR DESCRIPTION
**Pull request by rhoque-glx**
**Description**

**Issues that should be CLOSED by merge of this PR:**
* Fixes #

**Related issues that should be LINKED to this PR:**
* Connects #

### CODE REVIEW

File: traveler-ios-ui/TravelerKitUI/Models/ItineraryByDay.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+//
+//  Created by Rakin Hoque on 2019-12-04.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import Foundation
+import TravelerKit
+
+// TODO: Think of a way to make this more generic and if possible move into Core
+public class ItineraryByDay {
```
Comment (OmarPadierna) [inline traveler-ios-ui/TravelerKitUI/Models/ItineraryByDay.swift]: structs over classes.

File: traveler-ios-ui/TravelerKitUI/Views/FlightDetailCell.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//  FlightDetailCell.swift
+//  TravelerKitUI
+//
+//  Created by Rakin Hoque on 2019-12-04.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+
+class FlightDetailCell: UITableViewCell {
```
Comment (OmarPadierna) [inline traveler-ios-ui/TravelerKitUI/Views/FlightDetailCell.swift]: Bad naming, view can't be model aware. The cell shuold have a generic name with generic labels. The same is true for the nibs.

File: traveler-ios-ui/TravelerKitUI/Views/FlightSummaryCell.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//  ItineraryDetailsView.swift
+//  TravelerKitUI
+//
+//  Created by Rakin Hoque on 2019-11-29.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+
+class FlightSummaryCell: UITableViewCell {
```
Comment (OmarPadierna) [inline traveler-ios-ui/TravelerKitUI/Views/FlightSummaryCell.swift]: naming is model aware, the same for labels, this shouldnt happen.

File: traveler-ios-ui/TravelerKitUI/Views/DateRangeCell.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    func dateRangeCellDidUpdate(_ cell: DateRangeCell, endDate: Date)
+}
+
+open class DateRangeCell: UITableViewCell {
+    @IBOutlet weak var startDateField: UITextField!
+    @IBOutlet weak var endDateField: UITextField!
+    
+    weak var delegate: DateRangeCellDelegate?
+    
+    public var startDate: Date? {
```
Comment (OmarPadierna) [inline traveler-ios-ui/TravelerKitUI/Views/DateRangeCell.swift]: views should not be model aware. Views should be dumb, this logic does not belong here. It should be abstracted into a view controller and the cell should only care about receiving strings to put in its text fields.

File: traveler-ios-ui/TravelerKitUI/ModelExtensions/ItineraryItemType+PurchaseType.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+//  TravelerKitUI
+//
+//  Created by Rakin Hoque on 2019-12-06.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import Foundation
+import TravelerKit
+
+extension ItineraryItemType {
```
Comment (OmarPadierna) [inline traveler-ios-ui/TravelerKitUI/ModelExtensions/ItineraryItemType+PurchaseType.swift]: Add todo to delete when backend fixes its stuff

File: traveler-ios-ui/TravelerKitUI/Models/ItineraryByDay.swift (hunk 1 above)
Comment (rhoque-glx) [reply traveler-ios-ui/TravelerKitUI/Models/ItineraryByDay.swift]: None of our view models are structs, why does this one need to be?  Will require many changes to change this to a struct as is.

**Pull request merged**